src/Worker/Core/DurableTaskWorkerOptions.cs

```diff
@@ -93,6 +136,38 @@ public DataConverter DataConverter
     /// </remarks>
     public ConcurrencyOptions Concurrency { get; } = new();
 
+    /// <summary>
+    /// Gets or sets the versioning options for the Durable Task worker.
+    /// </summary>
+    /// <remarks>
+    /// Worker versioning controls how a worker will handle orchestrations of different versions. Defining both the
+    /// version of the worker, the versions that can be worked on, and what to do in case a version does not comply
+    /// with the given options.
+    /// </remarks>
+    public VersioningOptions Versioning
+    {
+        get => this.versioning;
+        set
+        {
+            if (value is not null)
+            {
+                this.IsVersioningSet = true;
+                this.Versioning.Version = value.Version;
+                this.Versioning.MatchStrategy = value.MatchStrategy;
+                this.Versioning.FailureStrategy = value.FailureStrategy;
+            }
+            else
+            {
+                this.IsVersioningSet = false;
```

Comment:
Rather than having a flag indicating whether versioning is set, wouldn't it be simpler to just make this property nullable, and use the null value to signal whether the property is set? I think it would be a lot simpler, and also give you the benefit of compiler checks everywhere this property gets consumed.

Reply:
I was following the style that was already present in the options: https://github.com/microsoft/durabletask-dotnet/blob/e32fa7a4ff9e569120957299508a58e88c115398/src/Worker/Core/DurableTaskWorkerOptions.cs#L105

If you prefer null I can do that instead.